src/pepr/operator/controllers/keycloak/authservice/authservice.ts

```diff
@@ -0,0 +1,120 @@
+import { Log, R } from "pepr";
+import { UDSConfig } from "../../../../config";
+import { Store } from "../../../common";
+import { UDSPackage } from "../../../crd";
+import { apiCall } from "../client-sync";
+import { Client } from "../types";
+import { updatePolicy } from "./authorization-policy";
+import { getAuthserviceConfig, operatorConfig, updateAuthServiceSecret } from "./config";
+import { Action, AuthServiceEvent, AuthserviceConfig, Chain } from "./types";
+
+export async function authservice(pkg: UDSPackage) {
+  // Get the list of clients from the package
+  const authServiceClients = R.filter(
+    sso => R.isNotNil(sso.enableAuthserviceSelector),
+    pkg.spec?.sso || [],
+  );
+
+  for (const client of authServiceClients) {
+    const name = `sso-client-${client.clientId}`;
+    const token = Store.getItem(name);
+    if (!token) {
+      throw new Error(`Failed to get token for client ${client.clientId}`);
+    }
+    const keycloakClient = await apiCall(client, "GET", token);
```

Comment:
Talked about this out of band  but basically the client secret is currently not 'stored' by the operator code which makes this HTTP call necessary.   It is definitely possible to return the client (with secret) from the keycloak() reconciliation and avoid this but it would require some refactoring of the keycloak client status process.

Reply:
I think that small refactor is worth it to reduce the dependency on yet another keycloak api call